c++/include/orc/Reader.hh

```diff
@@ -39,7 +39,7 @@ namespace orc {
   // classes that hold data members so we can maintain binary compatibility
   struct ReaderOptionsPrivate;
   struct RowReaderOptionsPrivate;
-
+  class KeyProvider;
```

Comment:
The `c++/include/orc` folder is public and files in it will be installed. If the `KeyProvider` is also public, please move its definition to the include folder as well. Otherwise, we should remove the related function from here.